### PR TITLE
ws reconnect

### PR DESCRIPTION
**Pull request by rileylnapier**
## Description

when we disconnect and reconnect we need to resubscribe so we need to keep track of all the subscriptions

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-provider/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   "dependencies": {
     "graphql": "^15.5.0",
     "react-use": "^17.2.1",
+    "reconnecting-websocket": "^4.4.0",
```
Comment (suhasdeshpande) [inline packages/react-provider/package.json:20]: this one was last updated 1 year ago?

File: packages/react-provider/src/ws.ts [hunk 2]
```diff
@@ -1,74 +1,91 @@
...
+    this.connection = new ReconnectingWebSocket(
+      `${this.url}/?clientKey=${this.clientKey}`
+    );
+
+    this.connection.onopen = this.onOpen.bind(this);
     this.connection.onmessage = this.onMessage.bind(this);
   }
 
+  onOpen(): void {
+    this.connected = true;
```
Comment (tk26) [inline packages/react-provider/src/ws.ts:35]: should we also do `this.connected = false` in `close()`?

Comment (rileylnapier) [reply packages/react-provider/src/ws.ts:35]: yeah prob, good idea

**Review by tk26 — APPROVED** [review]

File: packages/react-provider/package.json (hunk 1 above)
Comment (rileylnapier) [reply packages/react-provider/package.json:20]: looks like it yeah

Comment (suhasdeshpande) [reply packages/react-provider/package.json:20]: it also looks slightly heavy. but is it getting tree shaken? I am mostly curious if adding this dependency bloats our package or not.

Comment (scarney81) [reply packages/react-provider/package.json:20]: @suhasdeshpande this package is 2.6kb minified and gzipped https://bundlephobia.com/package/reconnecting-websocket@4.4.0

Comment (scarney81) [reply packages/react-provider/package.json:20]: It's also tree shakeable and has no dependencies.

Screen Shot 2021-06-30 at 3 28 52 PM

Comment (rileylnapier) [reply packages/react-provider/package.json:20]: curious, what made you think it was slighly heavy?   i don't see any indication of that.

**Review by tk26 — APPROVED** [review]

File: packages/react-provider/package.json (hunk 1 above)
Comment (suhasdeshpande) [reply packages/react-provider/package.json:20]: I didn't realize I was looking at unpacked size on npm. 
Screen Shot 2021-06-30 at 3 50 35 PM

Comment (rileylnapier) [reply packages/react-provider/package.json:20]: ahh, isee

Comment (scarney81) [reply packages/react-provider/package.json:20]: Unpacked size can be deceiving. Mobx is 3.84 MB unpacked but like 16.1kB minified and gzipped

**Pull request merged**